File: 21553 - Sai Nandan D/Week 1/1614_Maximum Nesting Depth of the Parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        nest = 0
        max_nest = 0
        for c in s:
            if c == '(':
                nest += 1
            elif c == ')':
                nest -= 1
            if nest > max_nest:
                max_nest = nest
            
        return max_nest

File: 21561 - Suvam Tamang/Week 1/1614_Maximum Nesting Depth of the Parentheses.java
class Solution {
    public int maxDepth(String s) {
        Stack <Character> stack = new Stack <> ();
        int maxDepth = 0;
        
        for(Character c : s.toCharArray()){
            if(c == '('){
                stack.push(c);
                maxDepth = Math.max(maxDepth, stack.size());
            }
            else if(c == ')'){
                stack.pop();
            }
        }
        return maxDepth;
    }
}
